src/BrightScriptDebugSession.ts

```diff
@@ -287,6 +306,157 @@ export class BrightScriptDebugSession extends DebugSession {
         }
     }
 
+    private componentLibraryPostfix: string = '__lib';
+
+    protected async prepareAndHostComponentLibraries(componentLibraries, componentLibrariesOutDir: string, port: number) {
+        if (componentLibraries && componentLibrariesOutDir) {
+            this.componentLibrariesOutDir = componentLibrariesOutDir;
+            this.componentLibrariesStagingDirPaths = [];
+            let replace = require('replace-in-file');
+            let libraryNumber: number = 0;
+
+            // #region Prepare the component libraries and create some name spacing for debugging
+            for (const componentLibrary of componentLibraries as any) {
+                libraryNumber ++;
+                componentLibrary.outDir = componentLibrariesOutDir;
+                let stagingFolder = await this.rokuDeploy.prepublishToStaging(componentLibrary);
+                let paths = glob.sync(path.join(stagingFolder, '**/*'));
+                let pathDetails: object = {};
+
+                // Add breakpoint lines to the staging files and before publishing
+                this.sendDebugLogLine('Adding stop statements for active breakpoints in Component Libraries');
+                this.convertBreakpointPaths(componentLibrary.rootDir, componentLibrary.rootDir);
+                await this.addBreakpointStatements(stagingFolder, componentLibrary.rootDir);
+
+                for (let filePath of paths) {
+                    //make the path relative (+1 for removing the slash)
+                    let relativePath = filePath.substring(stagingFolder.length + 1);
+                    let parsedPath = path.parse(relativePath);
+
+                    if (parsedPath.ext) {
+                        let originalRelativePath = relativePath;
+
+                        if (parsedPath.ext === '.brs') {
+                            // Create the new file name to be used
+                            let newFileName: string = `${parsedPath.name}${this.componentLibraryPostfix}${libraryNumber}${parsedPath.ext}`;
+                            relativePath = path.join(parsedPath.dir, newFileName);
+
+                            // Update all the file name references in the library to the new file names
+                            replace.sync({
```

Comment:
Use an async promise-returning version of this function

```typescript
await replace(...yourStuff);
```

Reply:
When trying to move to the async version I ran into another problem. I attempted to understand the issue but sadly I wasn't able to figure it out. This is following the import change in the comment above.
![image](https://user-images.githubusercontent.com/9591618/59079604-e774d900-88ba-11e9-8bec-4acb2083dea2.png)